### ISSUE TITLE
Extra logging to Matching algorithm

### ISSUE THREAD

**Issue by zabeen** [Anthony-Nolan/Atlas#1268]
Need to determine why Azure is restarting the host app during certain match requests.

Additional logs should be written during matching (Verbose log level), especially at places where batches or streams are being enumerated to lists, such during Matching Phase 2 filtering, and between the completion of sub-searches and the start of scoring.

Note: this should be done as a hotfix against `rc/2.0.2` of Atlas.

**Comment by zabeen** [comment]
Copied from the PR:
>rc/2.0.3 release to UAT-WMDA-ATLAS is blocked as I am stuck with the terraform changes to the service plan (#1264) - it's not working nicely.
So, to unblock the investigation, we should deploy the logging commit via a non-rc feature branch that is still based on stable/2.0.2 - and then we can decide if it should be released to prod at a later stage.

**Comment by zabeen** [comment]
### Investigation

- the extra logging has been merged into new branch, `spike/extra-matching-logging`, based on `rc/2.0.2` and will be released to UAT-WMDA-ATLAS for investigations.

**Comment by zabeen** [comment]
Changes live on the `spike` branch - will not be merging to `master` for now
